R/zchunk_LB134.Diet_Rfao.R

```diff
@@ -0,0 +1,384 @@
+      mutate(value = value * fhy_value) %>%
+      select(-fhy_value) ->
+      L134.Food_t_ctry_Dmnd_Y
+
+    # Match in GCAM regions, aggregate, and compute ratios from final historical year
+    # Original lines 112-125
+    L134.Food_t_ctry_Dmnd_Y %>%
+      left_join_error_no_match(select(iso_GCAM_regID, iso, GCAM_region_ID), by = "iso") %>%
+      group_by(GCAM_region_ID, GCAM_demand, year) %>%
+      summarise(value = sum(value)) ->
+      L134.Food_t_R_Dmnd_Y
+
+    L134.Food_t_R_Dmnd_Y %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      select(-year) %>%
+      rename(value_fhy = value) %>%
+      left_join(L134.Food_t_R_Dmnd_Y, by = c("GCAM_region_ID", "GCAM_demand")) %>%
+      mutate(value = value / value_fhy) %>%
+      select(-value_fhy) %>%
+      # Fill in missing values for regions that do not exist
+      ungroup %>%
+      complete(GCAM_region_ID = unique(iso_GCAM_regID$GCAM_region_ID),
+               nesting(GCAM_demand, year), fill = list(value = 1.0)) ->
+      L134.FoodRatio_R_Dmnd_Y
+
+    # Multiply ratios by the caloric demands in the final historical year
+    # Original lines 127-131
+    L134.pcFood_kcald_R_Dmnd_Y %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      select(-year) %>%
+      rename(value_fhy = value) %>%
+      right_join(L134.FoodRatio_R_Dmnd_Y, by = c("GCAM_region_ID", "GCAM_demand")) %>%
+      mutate(value = value * value_fhy) %>%
+      select(-value_fhy) %>%
+      bind_rows(filter(L134.pcFood_kcald_R_Dmnd_Y, year < max(HISTORICAL_YEARS)))->
+      L134.pcFood_kcald_R_Dmnd_Y
+
+    # Extend the projected diets to all years, assuming convergence year and demand levels
+    # Original lines 133-138
+    DIET_CONVERGENCE_YEAR <- 9999
+    CONGERENCE_KCALD_CROPS <- 2500
+    CONGERENCE_KCALD_MEAT <- 1000
+
+    L134.pcFood_kcald_R_Dmnd_Y %>%
+      complete(GCAM_region_ID, GCAM_demand,
+               year = unique(c(L134.pcFood_kcald_R_Dmnd_Y$year, HISTORICAL_YEARS, FUTURE_YEARS, DIET_CONVERGENCE_YEAR))) %>%
+      # fill in convergence year (9999) value
+      mutate(value = if_else(year == DIET_CONVERGENCE_YEAR,
+                             if_else(GCAM_demand == "crops", CONGERENCE_KCALD_CROPS, CONGERENCE_KCALD_MEAT),
+                             value)) %>%
+      # interpolate out to that convergence year
+      group_by(GCAM_region_ID, GCAM_demand) %>%
+      mutate(value = approx_fun(year, value)) %>%
+      filter(year %in% c(HISTORICAL_YEARS, FUTURE_YEARS)) ->
+      L134.pcFood_kcald_R_Dmnd_Y
+
+    # Alternative diet scenarios for the SSPs
+    # Original lines 140-
+    # this same logic happens over and over, so *use a function*
+    # scen - scenario
+    create_ssp_diet <- function(L102.pcgdp_thous90USD_Scen_R_Y, scen, demand, L134.pcFood_kcald_R_Dmnd_Y, A_FoodDemand_SSPs) {
+      if(demand == "crops") {
+        a <- 4545    # a and b are parameters for demand scaling function below
```

Comment:
I'd more call it a demand function. The function takes GDP per capita and computes food demand (in kcal/p/day).